core/include/gnuradio-4.0/Graph.hpp

```diff
@@ -701,6 +709,25 @@ class Graph : public gr::Block<Graph> {
         return message;
     }
 
+    std::optional<Message> propertyCallbackGraphGRC([[maybe_unused]] std::string_view propertyName, Message message) {
+        assert(propertyName == graph::property::kGraphGRC);
+
+        if (message.cmd == message::Command::Get) {
+            message.data = property_map{{"value", gr::saveGrc(*_pluginLoader, *this)}};
+        } else if (message.cmd == message::Command::Set) {
+            const auto& data        = message.data.value();
+            auto        yamlContent = std::get<std::string>(data.at("value"s));
+            *this                   = gr::loadGrc(*_pluginLoader, yamlContent);
```

Comment:
If I read this correctly, this will just replace the current graph with a new one which has been read from the yaml, right? It might do some cleanup in the old graph's destructor, but it cannot do blocking actions like correctly stopping the current graph, before removing it.

Question: Should it be/Is it possible to do this on a running graph, or does it have to be in stopped state? On the other hand this would make sending the message more complicated, since effectively you would need to trigger and wait for the state change before.

Reply:
While integrating in OD, I'm seeing a crash. I'll first add a unit-test for "Running Graph" case, so we capture the crash.

@ivan-cukic can comment on the rest